src/uucore/src/lib/features/perms.rs

```diff
@@ -307,14 +307,41 @@ impl ChownExecutor {
         }
 
         let ret = if self.matched(meta.uid(), meta.gid()) {
-            match wrap_chown(
+            // Use safe syscalls for root directory to prevent TOCTOU attacks
+            #[cfg(target_os = "linux")]
+            let chown_result = if path.is_dir() {
```

Comment:
I think you could use `cfg!` in the condition. It would allow you to get rid of the duplicate block starting on line 335.